accounting/accounting_cli_functions.py

```diff
@@ -13,10 +13,170 @@
 import argparse
 import time
 import sys
+import pwd
+import json
 
 import pandas as pd
 
 
+def count_ranks(ranks):
+    if "-" in ranks:
+        ranks_count = ranks.replace("-", ",").split(",")
+        return int(ranks_count[1]) - int(ranks_count[0]) + 1
+    else:
+        return int(ranks) + 1
+
+
+def get_username(userid):
+    try:
+        return pwd.getpwuid(userid).pw_name
+    except KeyError:
+        return str(userid)
+
+
+def get_uid(username):
+    try:
+        return pwd.getpwnam(username).pw_uid
+    except KeyError:
+        return str(username)
+
+
+def write_records_to_file(job_records, output_file):
+    op = open(output_file, "a+")
```

Comment:
This file descriptor is not closed currently.  I'd recommend switching to the context manager usage:

```suggestion
    with open(output_file, "a+") as op:
```